src/sa_web/static/js/views/map-view.js

```diff
@@ -230,7 +248,7 @@ var Shareabouts = Shareabouts || {};
     },
     addLandmarkLayerView: function(landmarkCollectionId) {
       return function(model) {
-        this.landmarkLayerViews[landmarkCollectionId][model.id] = new S.BasicLayerView({
+        this.layerViews[landmarkCollectionId][model.id] = new S.BasicLayerView({
```

Comment:
Considering that `this.layerViews` is now a hash of collections, instead of a single collection, the add/remove "layerView" methods are probably causing some bugs.